Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/common/taimaba/TaimabaEndpoints.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Kuroba - *chan browser https://github.com/Adamantcheese/Kuroba/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.github.adamantcheese.chan.core.site.common.taimaba;
+
+import com.github.adamantcheese.chan.core.model.Post;
+import com.github.adamantcheese.chan.core.model.orm.Board;
+import com.github.adamantcheese.chan.core.model.orm.Loadable;
+import com.github.adamantcheese.chan.core.site.common.CommonSite;
+
+import java.util.Locale;
+import java.util.Map;
+
+import okhttp3.HttpUrl;
+
+public class TaimabaEndpoints extends CommonSite.CommonEndpoints {
+    protected final CommonSite.SimpleHttpUrl root;
+    protected final CommonSite.SimpleHttpUrl sys;
+
+    public TaimabaEndpoints(CommonSite commonSite, String rootUrl, String sysUrl) {
+        super(commonSite); 
+        root = new CommonSite.SimpleHttpUrl(rootUrl);
+        sys = new CommonSite.SimpleHttpUrl(sysUrl);
+    }
+
+    @Override
+    public HttpUrl catalog(Board board) {
+        return root.builder().s(board.code).s("catalog.json").url();
+    }
+
+    @Override
+    public HttpUrl thread(Board board, Loadable loadable) {
+        return root.builder().s(board.code).s("res").s(loadable.no + ".json").url();
+    }
+
+    @Override
+    public HttpUrl thumbnailUrl(Post.Builder post, boolean spoiler, Map<String, String> arg) {
+        return sys.builder().s(post.board.code).s("thumb").s(arg.get("tim") + "s.jpg").url();
+    }
+
+    @Override
+    public HttpUrl imageUrl(Post.Builder post, Map<String, String> arg) {
+        return sys.builder().s(post.board.code).s("src").s(arg.get("tim") + "." + arg.get("ext")).url();
+    }
+
+    @Override
+    public HttpUrl icon(String icon, Map<String, String> arg) {
+        CommonSite.SimpleHttpUrl stat = sys.builder().s("static");
+
+        if (icon.equals("country")) {
+            stat.s("flags").s(arg.get("country_code").toLowerCase(Locale.ENGLISH) + ".png");
+        }
+
+        return stat.url();
+    }
+
+	@Override
+    public HttpUrl pages(Board board) {
+        return root.builder().s(board.code).s("threads.json").url();
+    }
+
+    @Override
+    public HttpUrl reply(Loadable loadable) {
+        return sys.builder().s(loadable.board.code).s("taimaba.pl").url();
```

Comment:
If you're having problems posting, you might want to see what data actually needs to be sent by looking at what network requests happen during a post, using some browser development feature.

Reply:
i've gatherd some data with Open Web Sniffer, but i'm pretty clueless how to implement it.
Read some of the things on the Overchan implementation. Maybe i can get some clues.
I don't really grasp it, but i attached the data i gathered below. I'll snoop around if i can find similarities in vichan implementations, i remeber somewhere reading field4 for example.

data when posting:
hostname:  boards.420chan.org
method:		POST
url:		/b/taimaba.pl
		type	main_frame

POST data:
board:	b
fart:	somenumber
field4:	test (comment)
file:	(empty)
link:	(empty)
name:	(empty)
		(empty)
parent:	somenumber (threadId)
password:	generated password
sage:	on (boolean?)
task: 	post
		
response headers:
status:		302
date:		<date>
set-cookie:	pwd[...]
location:	/b/
cf-cache-status:	DYNAMIC
some cloudfare stuff